middle_end/flambda2/terms/flambda_primitive.ml

```diff
@@ -755,7 +730,7 @@ let compare_unary_primitive p1 p2 =
       | Project_value_slot _ | Is_boxed_float | Is_flat_float_array | End_region
       | Obj_dup ),
       _ ) ->
-    Stdlib.compare (unary_primitive_numbering p1) (unary_primitive_numbering p2)
+    Stdlib.compare (constructor_tag p1) (constructor_tag p2)
```

Comment:
This is incorrect, as it will cause all nullary constructors to be compared as equal.